### DISCUSSION TITLE
Format code in case prettier rules are defined in .eslintrc

### DISCUSSION THREAD

**Discussion by pkropachev** [LazyVim/LazyVim#2310] [Q&A] [answered]
Hi!

I have the project where prettier is integrated with eslint and prettier rules are defined in `.eslintrc` (eslint-config-prettier and eslint-plugin-prettier are used).

I have the following configuration:

`lazy.lua`

```lua
{ import = "lazyvim.plugins.extras.linting.eslint" },
{ import = "lazyvim.plugins.extras.formatting.prettier" },
```

`plugins\formatting.lua`

```lua
return {
  -- conform formatter
  {
    "stevearc/conform.nvim",
    opts = {
      formatters_by_ft = {
        python = { "isort", "black", "autopep8" },
        cpp = { "clang_format" },
        c = { "clang_format" },
        ["*"] = { "trim_whitespace" },
      },
      formatters = {
        black = {
          prepend_args = { "-S" },
        },
      },
    },
  },
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = { eslint = {} },
      setup = {
        eslint = function()
          require("lazyvim.util").lsp.on_attach(function(client)
            if client.name == "eslint" then
              client.server_capabilities.documentFormattingProvider = true
            elseif client.name == "tsserver" then
              client.server_capabilities.documentFormattingProvider = false
            end
          end)
        end,
      },
    },
  },
}
```

When I open `ts` file then I see errors and warnings from linter according rules defined in config. Then I try to format it using `<leader>cf`. File is formatted, but not to according of prettier rules. If I execute "EslintFixAll" then file is formatted according to rules.

Could you please advise what can be wrong?

**Comment by dpetka2001** [comment]
From how I understand `conform` is working (and take it with a grain of salt, since I'm just an ordinary user like you), when you enable the extra `prettier` for formatting, `conform` uses the `prettier` binary to pass `stdin` and do the formatting. 

I believe in your case that you want to use `prettier` as a plugin to `eslint`, it's not able to understand what you really want to do. There's also an `eslint_d` formatter for `conform`, so maybe if you would configure that instead of `prettier`, you would be able to do what you want. You can also pass a `condition` in `opts.formatters[formatter]` for specific formatter in order to have more granular control over when a formatter should be activated and used for formatting (see [docs](https://www.lazyvim.org/plugins/formatting#conformnvim) where `dprint` uses the `condition` to only format when a `dprint.json` is present).

Unfortunately, I haven't used the workflow you're trying to use, so I believe a better place to ask about this would be the `conform` [repo](https://github.com/stevearc/conform.nvim)? Sorry for not being able to be of more help.

**Comment by pkropachev** [comment]
Thanks for the suggestion! I think problem is related to config (prettier rules in `.eslintrc`).

![PcAccess_eG8Hq0N50R](https://github.com/LazyVim/LazyVim/assets/34160907/596554cb-74b5-41ed-8fe1-983926703e5b)

![PcAccess_I3duyoYtRQ](https://github.com/LazyVim/LazyVim/assets/34160907/f925f45d-4c76-4d92-8a88-f7ad3a52c491)

![PcAccess_pGmNKxIJTD](https://github.com/LazyVim/LazyVim/assets/34160907/32ee54ce-fa7f-48fc-9022-81be0484a3a2)

In case I remove opts for `"neovim/nvim-lspconfig"` behaviour is not changed.

![PcAccess_KhMgseV2e0](https://github.com/LazyVim/LazyVim/assets/34160907/b9ad99ac-014d-442b-a900-07f6b31d30c1)

Perhaps I wrong understand [this recipe](https://www.lazyvim.org/configuration/recipes#add-eslint-and-use-it-for-formatting) from the documentation.

**Reply by dpetka2001** [reply, marked as answer]
That recipe is saying
> Important: make sure not to add prettier to null-ls, otherwise this won't work!

So I believe you shouldn't enable the `prettier` extra that adds `prettier` to `conform`. Again, I'm just guessing here, since I've never used this recipe on my end.

**Reply by pkropachev** [reply]
Ah, you're right. Finally I removed prettier from extras (remove from Mason as well).

![PcAccess_OPaAUExotW](https://github.com/LazyVim/LazyVim/assets/34160907/0bf7a2d9-68d5-4571-9e5f-4b5bedd1db03)

![PcAccess_mxQGt8aQVg](https://github.com/LazyVim/LazyVim/assets/34160907/1d68f58c-a0f5-45b6-ab13-c133027c1409)

LazyFormat looks like the following:

![PcAccess_hNDBxTMSsS](https://github.com/LazyVim/LazyVim/assets/34160907/be09fc6a-06ee-46a5-aad3-5a24660f9ab5)

And now formatting works correct.

Thanks!
